pkg/thanos/rules_test.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2022 The prometheus-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package thanos
+
+import (
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/go-kit/log"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/prometheus-operator/prometheus-operator/pkg/operator"
+)
+
+func TestMakeRulesConfigMapsThanos(t *testing.T) {
+	t.Run("shouldAcceptRuleWithValidPartialResponseStrategyValue", shouldAcceptRuleWithValidPartialResponseStrategyValue)
+	t.Run("shouldRejectRuleWithInvalidPartialResponseStrategyValue", shouldRejectRuleWithInvalidPartialResponseStrategyValue)
+	t.Run("shouldAcceptRuleWithPartialResponseStrategySet", shouldAcceptRuleWithPartialResponseStrategySet)
```

Comment:
Quite similar to `shouldAcceptRuleWithValidPartialResponseStrategyValue` so I suggest to remove this test.